### DISCUSSION TITLE
How to access a LARAGON project from another device in the same LAN using HTTPS?

### DISCUSSION THREAD

**Discussion by MGeurts** [leokhoa/laragon#549] [Q&A] [answered]
How can I access one of my LARAGON projects from a SMARTPHONE in the same LAN as the LARAGON development server in HTTPS (SSL)?

I need HTTPS access because my application needs to use the smartphone camera to scan barcodes or QR codes.
I need a local solution, no NGrok or internet based tunnel solution.

I already spent 2 days looking for a solution, without success.
Can someone please help? Thanks.

**Comment by araymoore** [comment]
I was doing the same thing using "https://[localnet.ip.address]/" - in modern browsers, you'll get a "safety warning" and typically have to click "advanced" and "Proceed" or "accept the risk and continue" etc. to continue to the site via https (I gave a very limited attempt at creating a cert for the site, but abandoned that path and just went with the "accept risk and continue" method). You should get the same options on the smartphone. Just put "https://" in front of the URL you're using to access the Laragon server and accept the risk and your camera connection should be enabled. Alternatively, you can also get a bluetooth barcode scanner off Amazon pretty cheap to connect to the smartphone and scan the codes with that (we're moving to that sort of solution this year in the project that we're using Laragon as the internal host). Good luck!

**Comment by MGeurts** [comment]
In the mean time, after digging for 4 days, I found the solution myself.
I'm using LARAGON to serve multiple LARAVEL projects at the same time and had a lot of problems because I tried to use the IP address of the LARAGON machine to access my projects. I solved the problem by using a local DNS server (Pi-hohle) in my LAN to resolve the domain names (like myproject-1.test or my-project-2.test) and configuring LARAVEL VITE the correct way for each project. I try to publish a blog post with the solution this week and will post it in this community. To be continued...

Thanks for the help.

**Comment by MGeurts** [comment, marked as answer]
This is the solution I found to make all (LARAVEL) projects served by LARAGON accessible in a LAN :

[https://www.kreaweb.be/laravel-11-vite-configuration/](https://www.kreaweb.be/laravel-11-vite-configuration/) 

Notice that my blog also contains a lot of tips & tricks on how to update different LARAGON components.

Thanks for the help.
